.github/workflows/clang-tidy.yml

```diff
@@ -9,7 +9,6 @@ on:
     paths:
       - .clang_tidy
       - .github/workflows/clang-tidy.yml
-  workflow_call:
```

Comment:
This is not called from anywhere, so let's remove it.

Reply:
It's called via `gh` in `clang-tidy_on_fix_merged.yml`.. i'm not 100% sure this `workflow_call` is needed but i suspect it is.

Reply:
We use `gh workflow run` which actually creates workflow_dispatch event.
I think someone tried to run `clang-tidy.yml` using `uses:` (workflow_call), but it doesn't work in a step, and this was left here.